tests/components/dlna_dms/conftest.py

```diff
@@ -0,0 +1,131 @@
+from homeassistant.const import CONF_DEVICE_ID, CONF_URL
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+MOCK_DEVICE_HOST: Final = "192.88.99.21"
+MOCK_DEVICE_BASE_URL: Final = f"http://{MOCK_DEVICE_HOST}"
+MOCK_DEVICE_LOCATION: Final = MOCK_DEVICE_BASE_URL + "/dms_description.xml"
+MOCK_DEVICE_NAME: Final = "Test Server Device"
+MOCK_DEVICE_TYPE: Final = "urn:schemas-upnp-org:device:MediaServer:1"
+MOCK_DEVICE_UDN: Final = "uuid:7bf34520-f034-4fa2-8d2d-2f709d4221ef"
+MOCK_DEVICE_USN: Final = f"{MOCK_DEVICE_UDN}::{MOCK_DEVICE_TYPE}"
+MOCK_SOURCE_ID: Final = "test_server_device"
+
+LOCAL_IP: Final = "192.88.99.1"
+EVENT_CALLBACK_URL: Final = "http://192.88.99.1/notify"
+
+NEW_DEVICE_LOCATION: Final = "http://192.88.99.7" + "/dmr_description.xml"
+
+
+@pytest.fixture
+def upnp_factory_mock() -> Iterable[Mock]:
+    """Mock the UpnpFactory class to construct DMS-style UPnP devices."""
+    with patch(
+        "homeassistant.components.dlna_dms.dms.UpnpFactory",
+        autospec=True,
+        spec_set=True,
+    ) as upnp_factory:
+        upnp_device = create_autospec(UpnpDevice, instance=True)
+        upnp_device.name = MOCK_DEVICE_NAME
+        upnp_device.udn = MOCK_DEVICE_UDN
+        upnp_device.device_url = MOCK_DEVICE_LOCATION
+        upnp_device.device_type = MOCK_DEVICE_TYPE
+        upnp_device.available = True
+        upnp_device.parent_device = None
+        upnp_device.root_device = upnp_device
+        upnp_device.all_devices = [upnp_device]
+        upnp_device.services = {
+            "urn:schemas-upnp-org:service:ContentDirectory:1": create_autospec(
+                UpnpService,
+                instance=True,
+                service_type="urn:schemas-upnp-org:service:ContentDirectory:1",
+                service_id="urn:upnp-org:serviceId:ContentDirectory",
+            ),
+            "urn:schemas-upnp-org:service:ConnectionManager:1": create_autospec(
+                UpnpService,
+                instance=True,
+                service_type="urn:schemas-upnp-org:service:ConnectionManager:1",
+                service_id="urn:upnp-org:serviceId:ConnectionManager",
+            ),
+        }
+        seal(upnp_device)
+        upnp_factory_instance = upnp_factory.return_value
+        upnp_factory_instance.async_create_device.return_value = upnp_device
+
+        yield upnp_factory_instance
+
+
+@pytest.fixture
+async def domain_data_mock(
+    hass: HomeAssistant, aioclient_mock, upnp_factory_mock
+) -> AsyncGenerator[DlnaDmsData, None]:
+    """Mock some global data used by this component.
+
+    This includes network clients and library object factories. Mocking it
+    prevents network use.
+
+    Yields the actual domain data, for ease of access
+    """
+    with patch(
+        "homeassistant.components.dlna_dms.dms.AiohttpSessionRequester", autospec=True
+    ):
+        yield get_domain_data(hass)
```

Comment:
We don't want to interact with integration details in the tests. We should aim at asserting the library mocks instead.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations